aiida_vasp/io/outcar.py

```diff
@@ -0,0 +1,63 @@
+"""
+Tools for parsing OUTCAR files
+"""
+
+
+class OutcarParser(object):
+    """
+    parse OUTCAR into a dictionary, which is supposed to be turned into ParameterData later.
+    """
+
+    def __init__(self, fname):
+        super(OutcarParser, self).__init__()
+        self.outcar_file = fname
+        self.properties = ['volume', 'energies', 'efermi']
+
+    @property
+    def output_dict(self):
+        """Parse the OUTCAR file and return the parsed values wrapped in a dictionary"""
+        output_dict = {}
+        for property_name in self.properties:
+            try:
+                # call the method corresponding to property_name.
+                result = getattr(self, '_read_' + property_name)()
```

Comment:
For each property you are going through the whole file. Are you sure the file is never large enough for that to be relevant in terms of performance? If so, this is fine, else I would suggest a structure like:
```
with open(self.outcar_file, 'r') as outcar_file_obj:
  for line in outcar_file_obj:
    self.read_properties()
```

where `read_properties()` loops over `self.properties` and calls `_read_*` methods which then only have to deal with a single line each.

Reply:
That is a valid point. The OUTCAR file can become quite large and there might be quite a lot of different properties, which should be parsed from it. I will definitely have to change this in one of the next iterations. There are some quantities though which are spread out over multiple lines. In this case the line parser would have remember previously read lines  or just take over control over the reading process for a while. I will think about how to solve this.